### PR TITLE
Refactor deploy

### PR DESCRIPTION
**Pull request by irainia**
This PR is to address the need to refactor `deploy` command.

To merge this PR, the following should be confirmed:

- [x] merge  #257

### CODE REVIEW

File: cmd/commands.go [hunk 1]
```diff
@@ -133,6 +134,63 @@ func New() *cli.Command {
...
+	return map[string]string{
+		"Authorization": fmt.Sprintf("Basic %s", a.Token),
+	}, nil
+}
+
+func (a *BasicAuthentication) RequireTransportSecurity() bool {
+	return false
+}
+
+func initClientConnection(l log.Logger, serverHost string, requestTimeout time.Duration) (
```
Comment (siddhanta-rath) [inline cmd/commands.go:165]: will it be a good idea to move all GRPC client-related code to a separate file?

Comment (irainia) [reply cmd/commands.go:165]: yeah, that could be a good idea. will try to check it and update it accordingly.

File: cmd/admin_build_instance.go [hunk 2]
```diff
@@ -81,24 +78,15 @@ func getInstanceBuildRequest(l log.Logger, jobName, inputDirectory, host, projec
...
 
-	dialTimeoutCtx, dialCancel := context.WithTimeout(context.Background(), OptimusDialTimeout)
-	defer dialCancel()
-
-	var conn *grpc.ClientConn
-	if conn, err = createConnection(dialTimeoutCtx, host); err != nil {
-		if errors.Is(err, context.DeadlineExceeded) {
-			l.Error(ErrServerNotReachable(host).Error())
-		}
+	ctx, conn, closeConn, err := initClientConnection(l, host, adminBuildInstanceTimeout)
```
Comment (sbchaos) [inline cmd/admin_build_instance.go:81]: Can you try pushing creation of dial context to createConnection. That context is only required at time of creating connection.

Also if err is context.DeadlineExceeded then return `ErrServerNotReachable` error.

File: cmd/commands.go (hunk 1 above)
Comment (irainia) [reply cmd/commands.go]: it's addressed

File: cmd/admin_build_instance.go (hunk 2 above)
Comment (irainia) [reply cmd/admin_build_instance.go:81]: concerns are received from separate call, will check and update accordingly

Comment (irainia) [reply cmd/admin_build_instance.go:81]: concerns are addressed and some bugs are fixed.

File: cmd/connectivity.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+) {
+	connection, err = createConnection(serverHost)
+	if err != nil {
+		return
+	}
+	reqCtx, reqCancel := context.WithTimeout(context.Background(), requestTimeout)
+
+	requestCtx = reqCtx
+	closeConnection = func() {
+		l.Info("Closing client connection")
```
Comment (sbchaos) [inline cmd/connectivity.go:91]: Is this for testing or do we need this info log ?

Comment (sbchaos) [inline cmd/connectivity.go:91]: Is this for testing or do we need this info log ?

**Review by sbchaos — APPROVED** [review]

File: cmd/connectivity.go (hunk 3 above)
Comment (irainia) [reply cmd/connectivity.go:91]: oh right, missed that one. should've removed it

Comment (irainia) [reply cmd/connectivity.go]: it is addressed

**Pull request merged**